task/mesosaurus-task.cpp

```diff
@@ -56,13 +58,33 @@ void* workerEntry(void* payload) {
   int allocated = 0;
   int chunkSize = 1024;
   int estimatedIterations = 1;
-
+  //decide if we are failing and what time we are going to fail at.
+  random_device rd;
+  //double lambda = (-1 * log(1-current_workload->fail_rate));
+  //exponential_distribution<> rng (lambda);
+  //double f = rng(rnd_gen);
+  //later we may want to add exponential_distribution failure support?
+  mt19937 rnd_gen( rd ());
+
+
+  uniform_real_distribution<double> dist(0,1);
+
+  bool fail = dist(rnd_gen) < current_workload->fail_rate;
+  long mid = (endTime+us_timestamp())/2;
+  long stdev = (endTime-us_timestamp())/2;
+  normal_distribution<> norm(mid, stdev);
+  long failureTime = norm(rnd_gen);
   printf("Worker %d: allocate %d bytes over work iterations\n",
-      current_workload->id, bytesToAllocate);
+    current_workload->id, bytesToAllocate);
 
   for (int iteration = 0; us_timestamp() < endTime; iteration++) {
     long start = us_timestamp();
 
+    if(fail && start > failureTime)
```

Comment:
Right now when fail is true it will(should) fail. The reason to check fail and the time is so that we can not have every failed task right when it starts. failureTime is computed using normal_distribution.